src/Core/Database.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Core;
+
+use PDO;
+
+class Database
+{
+    private $connection;
+    private $statement;
+
+    public function __construct(array $config = [], string $username = "root", string $password = "")
+    {
+        $dsn = "mysql:" . http_build_query($config, "", ";");
+        $this->connection = new PDO($dsn, $username, $password, [
+            PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
+            PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION
+        ]);
+    }
+
+    private function query(string $sql, array $params = [])
+    {
+        $this->statement = $this->connection->prepare($sql);
+        $this->statement->execute($params);
+    }
+
+    public function fetchAll(string $sql, array $params = [])
+    {
+        $this->query($sql, $params);
+        return $this->statement->fetchAll();
+    }
+
+    public function fetch(string $sql, array $params = [])
+    {
+        $this->query($sql, $params);
+        return $this->statement->fetch();
```

Comment:
`fetch` method which finds one record by its ID should throw `RecordNotFoundException` when no result returns. This will help in handling 404 error responses when for example finding a non-existing category by ID.
`RecordNotFoundException` class is not a PHP built-in class and you need to create a new exception class with this name.